src/api/upstreamApi.ts

```diff
@@ -0,0 +1,44 @@
+import { createProductPricingClient, ProductPricingClient } from './upstreams/productPricing';
+import { createContextfulHttpClient, HttpClient } from './httpClient';
+import { ForwardHeaders } from '../context';
+import * as config from '../config'
+
+/**
+ * An umbrella type that groups different upstream services, making it
+ * easy to inject in the Context.
+ *
+ * Then it can be used in functions like
+ * ```typescript
+ * { upstream } // injected into GraphQL resolve function for instance
+ * ...
+ * const quote = await upstream.underwriter.getQuote("id")
+ * ```
+ */
+export interface UpstreamApi {
+    productPricing: ProductPricingClient
+}
+
+export const createUpstreamApi = (
+    token: string,
+    headers: ForwardHeaders
+): UpstreamApi => {
+
+    const createHttpClient = (remotePathPrefix: string, localPort: number): HttpClient => {
+        const allHeaders = (headers as any) as { [key: string]: string }
+
+        let baseUrl: string = `${config.BASE_URL}${remotePathPrefix}`
+        if (config.UPSTREAM_MODE == "local") {
+            baseUrl = `http://localhost:${localPort}`
+            if (config.LOCAL_MEMBERID_OVERRIDE) {
+                allHeaders["Hedvig.token"] = config.LOCAL_MEMBERID_OVERRIDE as string
```

Comment:
I have no idea if this is the case, just to make sure
```suggestion
                allHeaders["Hedvig.Token"] = config.LOCAL_MEMBERID_OVERRIDE as string
```

Reply:
[HTTP headers are case insensitive](https://stackoverflow.com/a/5259004) - so it shouldn't matter :) I just chose the casing I saw most common in the backend for consistency.